### PR TITLE
Improve integration tests

### PR DESCRIPTION
**Pull request by lunny**
This PR tried to make integration tests locally easier and also fix some problem when parallel running different database integrations tests.

### CODE REVIEW

**Review by guillep2k — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

File: integrations/README.md [hunk 1]
```diff
@@ -27,31 +27,45 @@ make test-sqlite
 ## Run mysql integrations tests
 Setup a mysql database inside docker
 ```
-docker run -e "MYSQL_DATABASE=test" -e "MYSQL_ALLOW_EMPTY_PASSWORD=yes" --rm --name mysql mysql:5.7 #(just ctrl-c to stop db and clean the container) 
+docker run -e "MYSQL_DATABASE=test" -e "MYSQL_ALLOW_EMPTY_PASSWORD=yes" -p 3306:3306 --rm --name mysql mysql:5.7 #(just ctrl-c to stop db and clean the container) 
 ```
 Start tests based on the database container
 ```
-TEST_MYSQL_HOST="$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' mysql):3306" TEST_MYSQL_DBNAME=test TEST_MYSQL_USERNAME=root TEST_MYSQL_PASSWORD='' make test-mysql
+TEST_MYSQL_HOST=$(docker inspect -f 'localhost:3306' mysql) TEST_MYSQL_DBNAME=test TEST_MYSQL_USERNAME=root TEST_MYSQL_PASSWORD='' make test-mysql
```
Comment (sapk) [inline integrations/README.md]:
```suggestion
TEST_MYSQL_HOST=localhost:3306 TEST_MYSQL_DBNAME=test TEST_MYSQL_USERNAME=root TEST_MYSQL_PASSWORD='' make test-mysql
```

File: integrations/README.md [hunk 2]
```diff
@@ -27,31 +27,45 @@ make test-sqlite
...
 ## Run pgsql integrations tests
 Setup a pgsql database inside docker
 ```
-docker run -e "POSTGRES_DB=test" --rm --name pgsql postgres:9.5 #(just ctrl-c to stop db and clean the container) 
+docker run -e "POSTGRES_DB=test" -p 5432:5432 --rm --name pgsql postgres:9.5 #(just ctrl-c to stop db and clean the container) 
 ```
 Start tests based on the database container
 ```
-TEST_PGSQL_HOST=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' pgsql) TEST_PGSQL_DBNAME=test TEST_PGSQL_USERNAME=postgres TEST_PGSQL_PASSWORD=postgres make test-pgsql
+TEST_PGSQL_HOST=$(docker inspect -f 'localhost:5432' pgsql) TEST_PGSQL_DBNAME=test TEST_PGSQL_USERNAME=postgres TEST_PGSQL_PASSWORD=postgres make test-pgsql
```
Comment (sapk) [inline integrations/README.md]:
```suggestion
TEST_PGSQL_HOST=localhost:5432 TEST_PGSQL_DBNAME=test TEST_PGSQL_USERNAME=postgres TEST_PGSQL_PASSWORD=postgres make test-pgsql
```

File: integrations/README.md [hunk 3]
```diff
@@ -27,31 +27,45 @@ make test-sqlite
...
+```
+
+## Run mssql integrations tests
+Setup a mssql database inside docker
+```
+docker run -e "ACCEPT_EULA=Y" -e "MSSQL_PID=Standard" -e "SA_PASSWORD=MwantsaSecurePassword1" -p 1433:1433 --rm --name mssql microsoft/mssql-server-linux:latest #(just ctrl-c to stop db and clean the container) 
+```
+Start tests based on the database container
+```
+TEST_MSSQL_HOST=$(docker inspect -f 'localhost:1433' mssql) TEST_MSSQL_DBNAME=gitea_test TEST_MSSQL_USERNAME=sa TEST_MSSQL_PASSWORD=MwantsaSecurePassword1 make test-mssql
```
Comment (sapk) [inline integrations/README.md]:
```suggestion
TEST_MSSQL_HOST=localhost:1433 TEST_MSSQL_DBNAME=gitea_test TEST_MSSQL_USERNAME=sa TEST_MSSQL_PASSWORD=MwantsaSecurePassword1 make test-mssql
```

File: integrations/README_ZH.md [hunk 4]
```diff
@@ -26,31 +26,46 @@ make test-sqlite
 ## 如何使用 mysql 数据库进行集成测试
 首先在docker容器里部署一个 mysql 数据库
 ```
-docker run -e "MYSQL_DATABASE=test" -e "MYSQL_ALLOW_EMPTY_PASSWORD=yes" --rm --name mysql mysql:5.7 #(just ctrl-c to stop db and clean the container) 
+docker run -e "MYSQL_DATABASE=test" -e "MYSQL_ALLOW_EMPTY_PASSWORD=yes" -p 3306:3306 --rm --name mysql mysql:5.7 #(just ctrl-c to stop db and clean the container) 
 ```
 之后便可以基于这个数据库进行集成测试
 ```
-TEST_MYSQL_HOST="$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' mysql):3306" TEST_MYSQL_DBNAME=test TEST_MYSQL_USERNAME=root TEST_MYSQL_PASSWORD='' make test-mysql
+TEST_MYSQL_HOST=$(docker inspect -f 'localhost:3306' mysql) TEST_MYSQL_DBNAME=test TEST_MYSQL_USERNAME=root TEST_MYSQL_PASSWORD='' make test-mysql
```
Comment (sapk) [inline integrations/README_ZH.md]:
```suggestion
TEST_MYSQL_HOST=localhost:3306 TEST_MYSQL_DBNAME=test TEST_MYSQL_USERNAME=root TEST_MYSQL_PASSWORD='' make test-mysql
```

File: integrations/README_ZH.md [hunk 5]
```diff
@@ -26,31 +26,46 @@ make test-sqlite
...
 ## 如何使用 pgsql 数据库进行集成测试
 同上，首先在 docker 容器里部署一个 pgsql 数据库
 ```
-docker run -e "POSTGRES_DB=test" --rm --name pgsql postgres:9.5 #(just ctrl-c to stop db and clean the container) 
+docker run -e "POSTGRES_DB=test" -p 5432:5432 --rm --name pgsql postgres:9.5 #(just ctrl-c to stop db and clean the container) 
 ```
 之后便可以基于这个数据库进行集成测试
 ```
-TEST_PGSQL_HOST=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' pgsql) TEST_PGSQL_DBNAME=test TEST_PGSQL_USERNAME=postgres TEST_PGSQL_PASSWORD=postgres make test-pgsql
+TEST_PGSQL_HOST=$(docker inspect -f 'localhost:5432' pgsql) TEST_PGSQL_DBNAME=test TEST_PGSQL_USERNAME=postgres TEST_PGSQL_PASSWORD=postgres make test-pgsql
```
Comment (sapk) [inline integrations/README_ZH.md]:
```suggestion
TEST_PGSQL_HOST=localhost:5432 TEST_PGSQL_DBNAME=test TEST_PGSQL_USERNAME=postgres TEST_PGSQL_PASSWORD=postgres make test-pgsql
```

File: integrations/README_ZH.md [hunk 6]
```diff
@@ -26,31 +26,46 @@ make test-sqlite
...
+
+其它数据库(把 MSSQL 替换为 MYSQL, MYSQL8, PGSQL):
+
 ```
-go test -c code.gitea.io/gitea/integrations \
-  -o integrations.sqlite.test -tags 'sqlite' &&
-  GITEA_ROOT="$GOPATH/src/code.gitea.io/gitea" \
-  GITEA_CONF=integrations/sqlite.ini ./integrations.sqlite.test \
-  -test.v -test.run GPG
+TEST_MSSQL_HOST=$(docker inspect -f 'localhost:1433' mssql) TEST_MSSQL_DBNAME=test TEST_MSSQL_USERNAME=sa TEST_MSSQL_PASSWORD=MwantsaSecurePassword1 make test-mssql#GPG
```
Comment (sapk) [inline integrations/README_ZH.md]:
```suggestion
TEST_MSSQL_HOST=localhost:1433 TEST_MSSQL_DBNAME=test TEST_MSSQL_USERNAME=sa TEST_MSSQL_PASSWORD=MwantsaSecurePassword1 make test-mssql#GPG
```

**Review by sapk — CHANGES_REQUESTED** [review]
docker inspect command are not needed anymore.

File: Makefile [hunk 7]
```diff
@@ -204,94 +204,140 @@ test-sqlite\#%: integrations.sqlite.test
...
 		-e 's|{{TEST_MYSQL_USERNAME}}|${TEST_MYSQL_USERNAME}|g' \
 		-e 's|{{TEST_MYSQL_PASSWORD}}|${TEST_MYSQL_PASSWORD}|g' \
 			integrations/mysql.ini.tmpl > integrations/mysql.ini
+
+.PHONY: test-mysql
+test-mysql: integrations.mysql.test generate-ini-mysql
+	GITEA_ROOT=${CURDIR} GITEA_CONF=integrations/mysql.ini ./integrations.mysql.test
+
+.PHONY: test-mysql\#%
+test-mysql\#%: integrations.mysql.test
```
Comment (sapk) [inline Makefile]:
```suggestion
test-mysql\#%: integrations.mysql.test generate-ini-mysql
```

File: Makefile [hunk 8]
```diff
@@ -204,94 +204,140 @@ test-sqlite\#%: integrations.sqlite.test
...
 		-e 's|{{TEST_MYSQL8_USERNAME}}|${TEST_MYSQL8_USERNAME}|g' \
 		-e 's|{{TEST_MYSQL8_PASSWORD}}|${TEST_MYSQL8_PASSWORD}|g' \
 			integrations/mysql8.ini.tmpl > integrations/mysql8.ini
+
+.PHONY: test-mysql8
+test-mysql8: integrations.mysql8.test generate-ini-mysql8
+	GITEA_ROOT=${CURDIR} GITEA_CONF=integrations/mysql8.ini ./integrations.mysql8.test
+
+.PHONY: test-mysql8\#%
+test-mysql8\#%: integrations.mysql8.test
```
Comment (sapk) [inline Makefile]:
```suggestion
test-mysql8\#%: integrations.mysql8.test generate-ini-mysql8
```

File: Makefile [hunk 9]
```diff
@@ -204,94 +204,140 @@ test-sqlite\#%: integrations.sqlite.test
...
-	GITEA_ROOT=${CURDIR} GITEA_CONF=integrations/mysql.ini ./migrations.test
+.PHONY: test-pgsql
+test-pgsql: integrations.pgsql.test generate-ini-pgsql
+	GITEA_ROOT=${CURDIR} GITEA_CONF=integrations/pgsql.ini ./integrations.pgsql.test
 
-.PHONY: test-mysql8
-test-mysql8: integrations.test generate-ini
-	GITEA_ROOT=${CURDIR} GITEA_CONF=integrations/mysql8.ini ./integrations.test
+.PHONY: test-pgsql\#%
+test-pgsql\#%: integrations.pgsql.test
```
Comment (sapk) [inline Makefile]:
```suggestion
test-pgsql\#%: integrations.pgsql.test generate-ini-pgsql
```

File: Makefile [hunk 10]
```diff
@@ -204,94 +204,140 @@ test-sqlite\#%: integrations.sqlite.test
...
+			integrations/mssql.ini.tmpl > integrations/mssql.ini
 
 .PHONY: test-mssql
-test-mssql: integrations.test generate-ini
-	GITEA_ROOT=${CURDIR} GITEA_CONF=integrations/mssql.ini ./integrations.test
+test-mssql: integrations.mssql.test generate-ini-mssql
+	GITEA_ROOT=${CURDIR} GITEA_CONF=integrations/mssql.ini ./integrations.mssql.test
+
+.PHONY: test-mssql\#%
+test-mssql\#%: integrations.mssql.test
```
Comment (sapk) [inline Makefile]:
```suggestion
test-mssql\#%: integrations.mssql.test generate-ini-mssql
```

**Review by sapk — CHANGES_REQUESTED** [review]
please add required step `generate-ini-mysql`

File: integrations/README.md [hunk 11]
```diff
@@ -27,31 +27,45 @@ make test-sqlite
...
-  GITEA_ROOT="$GOPATH/src/code.gitea.io/gitea" \
-  GITEA_CONF=integrations/sqlite.ini ./integrations.sqlite.test \
-  -test.v -test.run GPG
+make test-sqlite#GPG
 ```
 
+For other databases(replace MSSQL to MYSQL, MYSQL8, PGSQL):
+
+```
+TEST_MSSQL_HOST=$(docker inspect -f 'localhost:1433' mssql) TEST_MSSQL_DBNAME=test TEST_MSSQL_USERNAME=sa TEST_MSSQL_PASSWORD=MwantsaSecurePassword1 make test-mssql#GPG
```
Comment (sapk) [inline integrations/README.md]:
```suggestion
TEST_MSSQL_HOST=localhost:1433 TEST_MSSQL_DBNAME=test TEST_MSSQL_USERNAME=sa TEST_MSSQL_PASSWORD=MwantsaSecurePassword1 make test-mssql#GPG
```

File: integrations/README_ZH.md [hunk 12]
```diff
@@ -26,31 +26,46 @@ make test-sqlite
...
+```
+
+## Run mssql integrations tests
+同上，首先在 docker 容器里部署一个 mssql 数据库
+```
+docker run -e "ACCEPT_EULA=Y" -e "MSSQL_PID=Standard" -e "SA_PASSWORD=MwantsaSecurePassword1" -p 1433:1433 --rm --name mssql microsoft/mssql-server-linux:latest #(just ctrl-c to stop db and clean the container) 
+```
+之后便可以基于这个数据库进行集成测试
+```
+TEST_MSSQL_HOST=$(docker inspect -f 'localhost:1433' mssql) TEST_MSSQL_DBNAME=gitea_test TEST_MSSQL_USERNAME=sa TEST_MSSQL_PASSWORD=MwantsaSecurePassword1 make test-mssql
```
Comment (sapk) [inline integrations/README_ZH.md]:
```suggestion
TEST_MSSQL_HOST=localhost:1433 TEST_MSSQL_DBNAME=gitea_test TEST_MSSQL_USERNAME=sa TEST_MSSQL_PASSWORD=MwantsaSecurePassword1 make test-mssql
```

**Review by sapk — CHANGES_REQUESTED** [review]
I missed those two lines also ...

**Comment by lunny** [comment]
All done.

**Review by sapk — APPROVED** [review]

**Pull request merged**